lib/crawler.js

```diff
@@ -248,70 +293,87 @@ Crawler.prototype._pushToQueue = function _pushToQueue (options) {
 	self.globalOnlyOptions.forEach(globalOnlyOption => delete options[globalOnlyOption]);
 
 	// If duplicate skipping is enabled, avoid queueing entirely for URLs we already crawled
-	if (!self.options.skipDuplicates){
+	if (!self.options.skipDuplicates) {
 		self._schedule(options);
 		return;
 	}
 
 	self.seen.exists(options, options.seenreq).then(rst => {
-		if(!rst){
+		if (!rst) {
 			self._schedule(options);
 		}
 	}).catch(e => log('error', e));
 };
 
-Crawler.prototype._schedule = function _scheduler(options){
+Crawler.prototype._schedule = function _scheduler(options) {
 	var self = this;
-	self.emit('schedule',options);
+	//NOTE this will be used to add proxy outside the class
+	self.emit('schedule', options);
 
-	self.limiters.key(options.limiter||'default').submit(options.priority,function(done, limiter){
-		options.release = function(){ done();self.emit('_release'); };
-		if(!options.callback)
+	self.limiters.key(options.limiter || 'default').submit(options.priority, function (done, limiter) {
+		options.release = function () { done(); self.emit('_release'); };
+		if (!options.callback)
 			options.callback = options.release;
 
 		if (limiter) {
 			self.emit('limiterChange', options, limiter);
 		}
 
 		if (options.html) {
-			self._onContent(null, options, {body:options.html,headers:{'content-type':'text/html'}});
+			self._onContent(null, options, { body: options.html, headers: { 'content-type': 'text/html' } });
 		} else if (typeof options.uri === 'function') {
-			options.uri(function(uri) {
+			options.uri(function (uri) {
 				options.uri = uri;
 				self._buildHttpRequest(options);
 			});
 		} else {
 			self._buildHttpRequest(options);
 		}
 	});
+
+};
+
+Crawler.prototype._clearHttp2Session = function _clearHttp2Session() {
+	log('debug', `Crawler clear all ${Object.keys(this.http2Connections).length} http2 connections`);
+	Object.keys(this.http2Connections).forEach(hostName => {
+		this._closeAndDeleteHttp2Session(hostName);
+		log('debug', `http2 connection to ${hostName} closed`);
+	});
+};
+
+Crawler.prototype._closeAndDeleteHttp2Session = function _closeAndDeleteHttp2Session(targetHost) {
+	if (this.http2Connections[targetHost]) {
+		this.http2Connections[targetHost].close();
```

Comment:
Will the connection be destroyed or closed? In the situation we have trouble to call `call`.